src/service_aggregator.py

```diff
@@ -251,7 +272,11 @@ async def collect_callback_responses(guid, num_queries):
 
     # Don't spend any more time than this assembling messages
     #OVERALL_TIMEOUT = timedelta(hours=1)  # 1 hour
-    OVERALL_TIMEOUT = timedelta(minutes=3)  # 3 minutes
+    logger.info(f"{guid}: params: {params}")
+    if "timeout_seconds" in params:
+        OVERALL_TIMEOUT = timedelta(seconds=params["timeout_seconds"])
```

Comment:
type check on timeout_seconds.

Reply:
I followed the flow around and at this point, params is either an empty dict or the timeout_seconds type has been checked already. So I simplified this logic a little.